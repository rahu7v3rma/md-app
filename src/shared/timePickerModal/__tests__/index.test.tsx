import moment from 'moment';
import React from 'react';
import renderer, { act } from 'react-test-renderer';

import TimePickerModal, {
    TimePickerModalElement
} from '@/shared/timePickerModal';

describe('<TimePickerModal />', () => {
    it('Time picker modal Snapshot Testing with Jest', async () => {
        const ref = React.createRef<TimePickerModalElement>();
        const tree = renderer.create(
            <TimePickerModal
                ref={ref}
                selectedValue={moment('2023-10-01T00:00:00.000Z').toDate()}
            />
        );
        await act(() => ref.current?.openTimePickerModal());

        expect(tree).toMatchSnapshot();
    });

    it('should render without errors', async () => {
        const ref = React.createRef<TimePickerModalElement>();
        const tree = renderer.create(
            <TimePickerModal
                ref={ref}
                selectedValue={moment('2023-10-01T00:00:00.000Z').toDate()}
            />
        );
        await act(() => ref.current?.openTimePickerModal());

        expect(tree.root).toBeTruthy();
    });

    it('should open the modal using the ref', async () => {
        const ref = React.createRef<TimePickerModalElement>();
        const tree = renderer.create(
            <TimePickerModal
                ref={ref}
                selectedValue={moment('2023-10-01T00:00:00.000Z').toDate()}
            />
        );
        await act(() => ref.current?.openTimePickerModal());

        const modal = tree.root.findByProps({
            testID: 'timePickerModal'
        }).props;

        expect(modal.isVisible).toBeTruthy();
    });

    it('should close the modal by swiping down', async () => {
        const ref = React.createRef<TimePickerModalElement>();
        const mockOnDismiss = jest.fn();
        const tree = renderer.create(
            <TimePickerModal
                disMissModal={mockOnDismiss}
                ref={ref}
                selectedValue={new Date()}
            />
        );

        await act(async () => {
            ref.current?.openTimePickerModal();
        });
        await act(() => {
            const modal = tree.root.findByProps({
                testID: 'timePickerModal'
            });
            modal.props.onSwipeComplete();
        });

        const modal = tree.root.findByProps({
            testID: 'timePickerModal'
        }).props;

        expect(mockOnDismiss).toBeCalledTimes(1);
        expect(modal.isVisible).toBeFalsy();
    });

    it('should select a Time and trigger the onSelect callback', async () => {
        const selectedDateString = moment(
            '2023-01-01 10:10',
            'YYYY-MM-DD hh:mm'
        ).toDate();
        const onTimeSelected = jest.fn().mockImplementation((date: Date) => {
            expect(moment(date).format('YYYY-MM-DD hh:mm')).toEqual(
                moment(selectedDateString).format('YYYY-MM-DD hh:mm')
            );
        });
        const ref = React.createRef<TimePickerModalElement>();
        const tree = renderer.create(
            <TimePickerModal
                ref={ref}
                selectedValue={selectedDateString}
                onSelect={onTimeSelected}
            />
        );

        await act(() => ref.current?.openTimePickerModal());

        const saveBtn = tree.root.findByProps({
            testID: 'saveBtn'
        }).props;

        await act(() => saveBtn.onPress());

        expect(onTimeSelected).toHaveBeenCalledTimes(1);
    });

    it('should display the correct title based on the locale', async () => {
        const ref = React.createRef<TimePickerModalElement>();
        const tree = renderer.create(
            <TimePickerModal
                ref={ref}
                selectedValue={new Date()}
                locale="en_GB"
            />
        );

        await act(() => ref.current?.openTimePickerModal());
        const title = tree.root.findByProps({
            testID: 'title'
        });

        expect(title.props.children).toEqual('Select Fast Duration');
    });

    it('should display the save button', async () => {
        const ref = React.createRef<TimePickerModalElement>();
        const tree = renderer.create(
            <TimePickerModal
                ref={ref}
                selectedValue={moment('2023-10-01').startOf('d').toDate()}
            />
        );
        await act(() => ref.current?.openTimePickerModal());

        const saveBtn = tree.root.findByProps({
            testID: 'saveBtn'
        });

        expect(saveBtn).toBeTruthy();
    });
});
