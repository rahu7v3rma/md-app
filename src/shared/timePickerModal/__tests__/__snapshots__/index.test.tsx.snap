// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimePickerModal /> Time picker modal Snapshot Testing with Jest 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  onSwipeComplete={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={false}
  supportedOrientations={
    [
      "portrait",
      "landscape",
    ]
  }
  swipeDirection="down"
  swipeThreshold={100}
  testID="timePickerModal"
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    collapsable={false}
    deviceHeight={null}
    deviceWidth={null}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onSwipeComplete={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    style={
      {
        "flex": 1,
        "justifyContent": "center",
        "left": 0,
        "margin": 0,
        "top": 0,
        "transform": [
          {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      [
        "portrait",
        "landscape",
      ]
    }
    swipeDirection="down"
    swipeThreshold={100}
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "height": 1334,
          "justifyContent": "flex-end",
          "position": "absolute",
          "transform": [
            {
              "translateY": 0,
            },
          ],
          "width": 750,
          "zIndex": 1000,
        }
      }
      testID="timerPickerContainer"
    >
      <View
        onTouchEnd={[Function]}
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "flex-end",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#F5F8FB",
              "borderRadius": 3,
              "height": 6,
              "marginBottom": 12,
              "opacity": 1,
              "width": 60,
            }
          }
        />
      </View>
      <View
        style={
          {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 48,
            "marginBottom": 32,
            "marginHorizontal": 8,
            "overflow": "scroll",
            "paddingBottom": 48,
            "paddingHorizontal": 20,
            "paddingTop": 48,
            "width": "94%",
          }
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          style={
            [
              {
                "fontFamily": "Poppins",
              },
              {
                "height": 56,
                "textAlign": "center",
              },
              {
                "color": "#271A51",
                "fontSize": 14,
                "fontWeight": "700",
                "lineHeight": 22,
              },
            ]
          }
          testID="title"
        >
          Select Start Time
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "marginBottom": 30,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              adjustsFontSizeToFit={true}
              style={
                [
                  {
                    "fontFamily": "Poppins",
                  },
                  {
                    "marginRight": 72,
                  },
                  {
                    "color": "#271A51",
                    "fontSize": undefined,
                    "fontWeight": undefined,
                    "lineHeight": undefined,
                  },
                ]
              }
            >
              :
            </Text>
          </View>
          <RNDatePicker
            androidVariant="iosClone"
            cancelText="Cancel"
            confirmText="Confirm"
            date={1696118400000}
            is24hourSource="locale"
            minuteInterval={1}
            mode="time"
            onChange={[Function]}
            onDateChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "height": 216,
                  "width": 310,
                },
                undefined,
              ]
            }
            textColor="#271A51"
            theme="light"
            title="Select time"
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#72996B",
              "borderColor": "#72996B",
              "borderRadius": 16,
              "justifyContent": "center",
              "margin": 0,
              "opacity": 1,
              "paddingHorizontal": 15.5,
              "paddingVertical": 14,
              "width": "100%",
            }
          }
          testID="saveBtn"
        >
          <Text
            adjustsFontSizeToFit={true}
            style={
              [
                {
                  "fontFamily": "Poppins",
                },
                undefined,
                {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "fontWeight": "600",
                  "lineHeight": 22,
                },
              ]
            }
          >
            Save
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
